### PR TITLE
Fedex

### PR DESCRIPTION
**Pull request by danielmarbach**
@andreasohlund I decided to use Nancy but no IoC magic. Just simple ugly static thingy which allows to customize the simulator, hope you are fine with that.

### CODE REVIEW

**Comment by andreasohlund** [comment]
Nancy is fine, low tech and simple ftw

File: FedEx.Gateway/App.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (andreasohlund) [inline FedEx.Gateway/App.config]: Do we need the config file?

File: FedEx.Simulator/Program.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+            BehaviorHolder.Behavior = behaviors[0];
+
+            using (var host = new NancyHost(new Uri("http://localhost:8888")))
+            {
+                host.Start();
+                Console.WriteLine("What behavior would you like?");
+                Console.WriteLine("[1] TimeoutException");
+                Console.WriteLine("[2] Success (default)");
+                Console.WriteLine("[3] Take longer than logical timeout");
```
Comment (andreasohlund) [inline FedEx.Simulator/Program.cs]: Perhaps a "Delay with X seconds" instead?

File: FedEx.Simulator/TakeLonger.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace FedEx.Simulator
+{
+    using System.Threading;
+
+    public class TakeLonger : FedexBehavior
+    {
+        public void Simulate()
+        {
+            Thread.Sleep(15000);
```
Comment (andreasohlund) [inline FedEx.Simulator/TakeLonger.cs]: Like mentioned above have the users configure this value?

Comment (danielmarbach) [inline FedEx.Simulator/TakeLonger.cs]: Yeah that was the plan. I thought about a simple constants file to include

On Tuesday, November 24, 2015, Andreas Öhlund notifications@github.com
wrote:

> In FedEx.Simulator/TakeLonger.cs
> https://github.com/Particular/SagaMasterClass/pull/3#discussion_r45777937
> :
> 
> > @@ -0,0 +1,12 @@
> > +﻿namespace FedEx.Simulator
> > +{
> > -    using System.Threading;
> >   +
> > -    public class TakeLonger : FedexBehavior
> > -    {
> > -        public void Simulate()
> > -        {
> > -            Thread.Sleep(15000);
> 
> Like mentioned above have the users configure this value?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Particular/SagaMasterClass/pull/3/files#r45777937.

## 

_#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more http://go.particular.net/l/27442/2015-08-21/5d4wzp →_

Comment (andreasohlund) [inline FedEx.Simulator/TakeLonger.cs]: Just have them specify it in the command: > delay 15s

Comment (danielmarbach) [inline FedEx.Simulator/TakeLonger.cs]: Ok. And then pass it into the http service over the payload?

On Tuesday, November 24, 2015, Andreas Öhlund notifications@github.com
wrote:

> In FedEx.Simulator/TakeLonger.cs
> https://github.com/Particular/SagaMasterClass/pull/3#discussion_r45785322
> :
> 
> > @@ -0,0 +1,12 @@
> > +﻿namespace FedEx.Simulator
> > +{
> > -    using System.Threading;
> >   +
> > -    public class TakeLonger : FedexBehavior
> > -    {
> > -        public void Simulate()
> > -        {
> > -            Thread.Sleep(15000);
> 
> Just have them specify it in the command: > delay 15s
> … <#1513b0c4fb6c7918_>
> On Tue, Nov 24, 2015 at 8:32 PM, Daniel Marbach <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> FedEx.Simulator/TakeLonger.cs <#3 (comment)
> https://github.com/Particular/SagaMasterClass/pull/3#discussion_r45782801>
> : > @@ -0,0 +1,12 @@ > +﻿namespace FedEx.Simulator > +{ > + using
> System.Threading; > + > + public class TakeLonger : FedexBehavior > + { > +
> public void Simulate() > + { > + Thread.Sleep(15000); Yeah that was the
> plan. I thought about a simple constants file to include …
> <#1513afbb3e7df205_> On Tuesday, November 24, 2015, Andreas Öhlund <
> notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> FedEx.Simulator/TakeLonger.cs <#3
> https://github.com/Particular/SagaMasterClass/pull/3 (comment) <#3
> (comment)
> https://github.com/Particular/SagaMasterClass/pull/3#discussion_r45777937>>
> : > @@ -0,0 +1,12 @@ > +﻿namespace FedEx.Simulator > +{ > + using
> System.Threading; > + > + public class TakeLonger : FedexBehavior > + { > +
> public void Simulate() > + { > + Thread.Sleep(15000); Like mentioned above
> have the users configure this value? — Reply to this email directly or view
> it on GitHub <
> https://github.com/Particular/SagaMasterClass/pull/3/files#r45777937>. --
> _#NSBcon 2015 • The "All About NServiceBus" conference is back \* *• Learn
> more http://go.particular.net/l/27442/2015-08-21/5d4wzp →_ — Reply to
> this email directly or view it on GitHub <
> https://github.com/Particular/SagaMasterClass/pull/3/files#r45782801>.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Particular/SagaMasterClass/pull/3/files#r45785322.

## 

_#NSBcon 2015 • The "All About NServiceBus" conference is back *
*• Learn more http://go.particular.net/l/27442/2015-08-21/5d4wzp →_

Comment (andreasohlund) [inline FedEx.Simulator/TakeLonger.cs]: Ah,I thought that the simulator had a console?

Comment (andreasohlund) [inline FedEx.Simulator/TakeLonger.cs]: Ie the delay is specified in the simulator it self?

**Comment by danielmarbach** [comment]
@andreasohlund you can pull in. I'll handle the Ups.Gateway later

**Pull request merged**